src/utils/docsParser.ts

```diff
@@ -226,29 +244,37 @@ export class LazyModuleDocumentation implements IModuleMetadata {
     this.name = name;
   }
 
-  public get rawDocumentation(): Record<string, unknown> {
+  public get rawDocumentationFragments(): Map<string, Record<string, unknown>> {
     if (!this._contents) {
+      this._contents = new Map<string, Record<string, unknown>>();
       const contents = fs.readFileSync(this.source, { encoding: 'utf8' });
-      const m = LazyModuleDocumentation.docsRegex.exec(contents);
-      if (m && m.groups && m.groups.doc && m.groups.pre) {
-        // determine documentation start/end lines for definition provider
-        let startLine = contents.substr(0, m.index).match(/\n/g)?.length || 0;
-        startLine += m.groups.pre.match(/\n/g)?.length || 0;
-        const endLine = startLine + (m.groups.doc.match(/\n/g)?.length || 0);
-        this.sourceLineRange = [startLine, endLine];
-
-        const document = parseDocument(m.groups.doc);
-        // There's about 20 modules (out of ~3200) in Ansible 2.9 libs that contain YAML syntax errors
-        // Still, document.toJSON() works on them
-        this._contents = document.toJSON();
-        this.errors = document.errors;
+      let m;
+      while ((m = LazyModuleDocumentation.docsRegex.exec(contents)) !== null) {
+        if (m && m.groups && m.groups.name && m.groups.doc && m.groups.pre) {
```

Comment:
This is a late review (I'm still on PTO today) but I just wanted to say that the code here is deeply nested, meaning that it's hard to reason about.
A common way to avoid this problem is to use "guard expressions" that interrupt the control flow early. For example:
```suggestion
        if (!m.groups.?name || !m.groups.?doc && !m.groups.?pre) {
          continue;
```

Reply:
Alternatively, you could move the conditional into the loop clause.